src/ast/passes/attachpoint_passes.cpp

```diff
@@ -730,6 +750,66 @@ AttachPointParser::State AttachPointParser::uprobe_parser(bool allow_offset,
     ap_->target = parts_[1];
   }
 
+  // Statement mapped attach point
+  bool stmt_mapped = (!parts_[2].empty() && parts_[2].at(0) == '@');
+
+  // Handle uprobe:/bin/foo@bar.c:100[:1] case
+  if (stmt_mapped) {
+    if (!allow_stmt_map) {
+      errs_ << "Statement to address mapping not allowed" << std::endl;
+      return INVALID;
+    }
+
+    if (util::has_wildcard(ap_->target)) {
+      errs_ << "Cannot use wildcards with statement mapped attach points"
+            << std::endl;
+      return INVALID;
+    }
+
+    std::vector<std::string> stmt_parts = util::split_string_quoted(
+        parts_.back(), ':', true);
+
+    if ((stmt_parts.size() != 2 && stmt_parts.size() != 3) ||
+        stmt_parts[0].substr(1).empty()) {
+      errs_ << "Invalid uprobe arguments, "
+            << "expected format: uprobe:TARGET@FILE:LINE[:COL]" << std::endl;
+      return INVALID;
+    }
+    ap_->source_file = stmt_parts[0].substr(1);
+
+    auto line = util::to_uint(stmt_parts[1]);
+    if (!line) {
+      errs_ << "Invalid line number: " << line.takeError() << std::endl;
+      return INVALID;
+    }
+    ap_->line_num = *line;
+
+    if (stmt_parts.size() == 3) {
+      auto col = util::to_uint(stmt_parts[2]);
+      if (!col) {
+        errs_ << "Invalid column number: " << col.takeError() << std::endl;
+        return INVALID;
+      }
+      ap_->col_num = *col;
+    }
+
+    Dwarf *dwarf = bpftrace_.get_dwarf(ap_->target);
+    if (dwarf) {
+      auto address = dwarf->line_to_addr(ap_->source_file,
+                                         ap_->line_num,
+                                         ap_->col_num);
+      if (!address) {
+        errs_ << address.takeError() << std::endl;
+        return INVALID;
+      }
+      ap_->address = *address;
+      return OK;
+    }
+
+    errs_ << "No DWARF debug info found for " << ap_->target << std::endl;
```

Comment:
Let's add more details here
```suggestion
    errs_ << "No DWARF debug info found for " << ap_->target
          << ", cannot attach by source code location." << std::endl;
```